web/ai_service.py

```
@@ -0,0 +1,141 @@
+            if settings.USE_OPENAI and self.openai_client:
+                response = self._get_openai_response(prompt, context)
+                if response:
+                    return self._process_markdown(response)
+            
+            return "I apologize, but I'm unable to process your request at the moment. Please try again later."
+            
+        except Exception as e:
+            print(f"Error in get_response: {str(e)}")
+            return "I apologize, but I encountered an error processing your request. Please try again later."
+
+    def _get_gemini_response(self, prompt: str, context: Optional[Dict[str, Any]] = None) -> Optional[str]:
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)